File: Core/src/burlton/core/code/util/StringUtil.java
package burlton.core.code.util;

public class StringUtil
{
	public static String convertOrdinalToText(int position)
	{
		if (position == -1)
		{
			return "";
		}
		
	    String[] suffixes = new String[] { "th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th" };
	    
	    int remainder = position % 100;
	    if (remainder == 11
	      || remainder == 12
	      || remainder == 13)
	    {
	    	//Special cases
	    	return position + "th";
	    }
	    else
	    {
	    	return position + suffixes[position%10];
	    }
	}
}

File: Core/src/burlton/core/code/util/StringUtil.kt
package burlton.core.code.util

object StringUtil
{
    fun convertOrdinalToText(position: Int): String
    {
        if (position < 0) return ""

        val suffixes = listOf("th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th")
        val remainder = position % 100
        return when (remainder)
        {
            11, 12, 13 -> "${position}th"
            else -> "${position}${suffixes[position % 10]}"
        }
    }
}